### PR TITLE
Describe `minicore` test auxiliary and directive

### PR DESCRIPTION
**Pull request by jieyouxu**
Context: [Real cross-compiling tests instead of `#![no_core]` silliness #130375](https://github.com/rust-lang/rust/issues/130375)
MCP: https://github.com/rust-lang/compiler-team/issues/786
Tracking issue: https://github.com/rust-lang/rust/issues/131485

Should only be merged after https://github.com/rust-lang/rust/pull/130693.

### CODE REVIEW

**Comment by jieyouxu** [comment]
Reminder for myself: this will need some more adjustments following rename from `//@ use-minicore` -> `//@ add-core-stubs`

**Comment by jieyouxu** [comment]
Updated directive name to `//@ add-core-stubs` and slightly adjusted docs for `minicore`.

**Review by DianQK — APPROVED** [review]
meow

File: src/tests/minicore.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+struct Meow;
+impl Copy for Meow {} // `Copy` here is provided by `minicore`
+
+// CHECK-LABEL: meow
+#[unsafe(no_mangle)]
+fn meow() {}
+```
+
+[minicore]: https://github.com/rust-lang/rust/tree/master/tests/auxiliary/minicore.rs
```
Comment (DianQK) [inline src/tests/minicore.md:54]: Ah. This should be 
```
[`minicore`]: https://github.com/rust-lang/rust/tree/master/tests/auxiliary/minicore.rs
```
?

Comment (jieyouxu) [reply src/tests/minicore.md:54]: Yeah I didn't notice this lol

**Pull request merged**